spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -497,15 +493,18 @@ private File retrieveOriginalFileFromHeader(Message<?> message) {
 
 	private File handleFileMessage(final File sourceFile, File tempFile, final File resultFile) throws IOException {
 		if (!FileExistsMode.APPEND.equals(this.fileExistsMode) && this.deleteSourceFiles) {
-			if (rename(sourceFile, resultFile)) {
+			try {
+				rename(sourceFile, resultFile);
 				return resultFile;
 			}
-			if (this.logger.isInfoEnabled()) {
-				this.logger.info(String.format("Failed to move file '%s'. Using copy and delete fallback.",
-						sourceFile.getAbsolutePath()));
+			catch (IOException e) {
+				if (this.logger.isInfoEnabled()) {
+					this.logger.info("Failed to move file '" + sourceFile + "'. Using copy and delete fallback.", e);
+				}
 			}
```

Comment:
I don't think there is a need for this any longer - `Files.move()` will copy the file if necessary - I tested it on OS X and `Files.move()` moved the file to another physical disk.

But I suppose it doesn't hurt to leave it here. WDYT?